### PR TITLE
Add NNCFBatchNorm1d, NNCFBatchNorm2d, NNCFBatchNorm3d.

### PR DESCRIPTION
**Pull request by asenina**
### Changes
Class `NNCFBatchNorm` was deleted. Classes `NNCFBatchNorm1d`, `NNCFBatchNorm2d`, `NNCFBatchNorm3d` were added.

### Reason for changes
This is necessary to test and support models with data of different shapes. It is related to #1089

### Related tickets
-

### Tests
-`tests/torch/pruning/filter_pruning/test_algo.py`

### CODE REVIEW

**Comment by daniil-lyakhov** [comment]
This pr is unblocking #1066 MNV3S075 pruning by Batchnorm1d support

**Comment by kshpv** [comment]
@vshampor can we merge all BatchNorm1d, BatchNorm2d, BatchNorm3d to only one NNCFBatchNorm?

**Review by alexsu52 — COMMENTED** [review]
Do you have tests for the pruning algorithm?

File: tests/torch/pruning/helpers.py [hunk 1]
```diff
@@ -215,25 +215,60 @@ def forward(self, x):
...
+#        x = self.conv_depthwise(x)
+#        x = self.conv2(x)
+#        x = self.bn2(x)
+#        x = self.relu(x)
+#        x = self.up(x)
+#        x = self.relu(x)
+#        x = self.conv3(x)
+#        x = x.view(1, -1)
+#        return x
+
```
Comment (daniil-lyakhov) [inline tests/torch/pruning/helpers.py]: Don't forget to remove this

**Comment by daniil-lyakhov** [comment]
Please update mobilenet_v3_block testcase in tests/torch/pruning/test_model_pruning_analysis.py:
 please add Batchnorm1d/2d to version with linear layer / conv in se block

**Comment by asenina** [comment]
Jenkins please retry a build

**Comment by asenina** [comment]
@alexsu52 @vshampor please take a look

**Review by alexsu52 — APPROVED** [review]
LGTM

**Review by daniil-lyakhov — APPROVED** [review]
👍

**Comment by asenina** [comment]
E2E build 813

File: nncf/torch/utils.py [hunk 2]
```diff
@@ -299,24 +299,43 @@ def default_distributed_unwrapper(model: nn.Module):
...
-    for name in legacy_bn_names:
-        tensor = state_dict_to_load.pop(name)
-        new_name = name.replace('BatchNorm2d', 'NNCFBatchNorm')
-        state_dict_to_load[new_name] = tensor
-
-    if legacy_bn_names:
-        warnings.warn('Legacy Batch Norm layer names was detected in checkpoint model state dict.'
-                      ' All occurrences of `BatchNorm2d` in nodes names was replaced by `NNCFBatchNorm`',
-                      category=DeprecationWarning)
+    legacy_bn2d_names, legacy_bn1d_names, legacy_bn3d_names, legacy_nncfbn_names = [], [], [], []
```
Comment (vshampor) [inline nncf/torch/utils.py:323]: Don't save the vertical space in this way, it reduces readability.

Comment (asenina) [reply nncf/torch/utils.py]: Removed.

File: nncf/torch/utils.py [hunk 3]
```diff
@@ -299,24 +299,43 @@ def default_distributed_unwrapper(model: nn.Module):
...
+            legacy_bn1d_names.append(name)
+        if 'BatchNorm3d' in name:
+            legacy_bn3d_names.append(name)
+        if 'NNCFBatchNorm' in name:
+            legacy_nncfbn_names.append(name)
+
+    rename_legacy_names_in_state_dict(state_dict_to_load, legacy_bn1d_names, 'BatchNorm1d', 'NNCFBatchNorm1d')
+    rename_legacy_names_in_state_dict(state_dict_to_load, legacy_bn2d_names, 'BatchNorm2d', 'NNCFBatchNorm2d')
+    rename_legacy_names_in_state_dict(state_dict_to_load, legacy_bn3d_names, 'BatchNorm3d', 'NNCFBatchNorm3d')
+    rename_legacy_names_in_state_dict(state_dict_to_load, legacy_nncfbn_names, 'NNCFBatchNorm', 'NNCFBatchNorm2d')
```
[the remark this answers is not in the document]
Comment (asenina) [reply nncf/torch/utils.py]: Done

File: nncf/torch/utils.py [hunk 4]
```diff
@@ -299,24 +299,43 @@ def default_distributed_unwrapper(model: nn.Module):
...
+                                      legacy_name: str,
+                                      new_name: str):
+
+    for name in legacy_names:
+        tensor = state_dict_to_load.pop(name)
+        new_key = name.replace(legacy_name, new_name) if not new_name in name else name
+        state_dict_to_load[new_key] = tensor
+
+    if legacy_names:
+        warnings.warn('Legacy Batch Norm layer names was detected in checkpoint model state dict.'
```
[the remark this answers is not in the document]
Comment (asenina) [reply nncf/torch/utils.py]: Done.

**Review by vshampor — APPROVED** [review]
